### PR TITLE
Added direct collection access to future collection item feature test

### PR DESCRIPTION
**Pull request by BrandonDusseau**
Further tests #5953 
/cc @jekyll/build

### CODE REVIEW

**Comment by BrandonDusseau** [comment]
@parkr It would have effectively made a duplicate of the test I modified. Since the scenario described matched the situation I was testing, I modified it. I can split it out into another test if you'd like.

File: features/collections.feature [hunk 1]
```diff
@@ -128,16 +128,18 @@ Feature: Collections
       puppies:
         output: false
     """
-    And I have a "foo.txt" file that contains "random static file"
+    And I have a "index.html" page that contains "Newest puppy: {% assign puppy = site.puppies.last %}{{ puppy.title }}"
```
Comment (parkr) [inline features/collections.feature]: I really thought `site.puppies.first` would be the newest.

Comment (BrandonDusseau) [reply features/collections.feature]: From my experience, apparently that is not the case. The documentation doesn't seem to specify.

**Comment by BrandonDusseau** [comment]
Split this into a separate test as previously suggested.

File: features/collections.feature [hunk 2]
```diff
@@ -139,6 +139,28 @@ Feature: Collections
     And the _site directory should exist
     And the "_site/puppies/fido.html" file should not exist
 
+  Scenario: Hidden collection with document with future date, accessed via Liquid
```
Comment (mattr-) [inline features/collections.feature]: This is nitpicky, but could you change this to `Hidden collection has document with future date, accessed via Liquid`

I had trouble parsing the scenario description until I read the full test.

Comment (BrandonDusseau) [reply features/collections.feature]: Sure, no problem. I was just keeping with the phrasing on the prior test. That'll be fixed in just a sec.

**Comment by mattr-** [comment]
Thanks! Appreciate you adding this test! 👍

**Comment by mattr-** [comment]
@jekyllbot: merge +dev

**Comment by parkr** [comment]
@mattr- I believe this is a failing test showing a behavior that isn’t working right. I see master as currently failing cucumber now on all ruby versions. Would you mind sending a patch to fix master and retry this patch with the proper fix to make the test pass?

**Comment by mattr-** [comment]
Oh! My bad! Why did I think this was passing earlier? 😞

**Comment by ashmaroli** [comment]
@mattr- do you have the time right now, or should I send the PR..?

**Pull request merged**